src/NServiceBus.Core.Tests/MessageMutators/MutatorRegistrationExtensionsTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace NServiceBus.Core.Tests.MessageMutators
+{
+    using System;
+    using System.Threading.Tasks;
+    using MessageMutator;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class MutatorRegistrationExtensionsTests
+    {
+        [Test]
+        public void Should_throw_ArgumentException_when_registering_non_mutator_type()
+        {
+            var endpointConfiguration = new EndpointConfiguration("test");
+
+            var exception = Assert.Throws<ArgumentException>(() => endpointConfiguration.RegisterMessageMutator(new object()));
+            StringAssert.Contains(
+                "The given instance is no valid message mutator. Implement one of the following mutator interfaces: NServiceBus.MessageMutator.IMutateIncomingMessages, NServiceBus.MessageMutator.IMutateIncomingTransportMessages, NServiceBus.MessageMutator.IMutateOutgoingMessages, NServiceBus.MessageMutator.IMutateOutgoingTransportMessages",
+                exception.Message);
+        }
+
+        [Theory]
```

Comment:
AFAIK this attribute isn't needed here. I just searched through the code, and I do see several other places where this has been done, but it's not needed. Theories in NUnit are something different than what we're doing. Using `TestCase` by itself should be enough.